lib/cocoapods-core/specification/linter.rb

```diff
@@ -434,6 +434,17 @@ def _validate_script_phases(s)
         end
       end
 
+      # Performs validations related to the `on_demand_resources` attribute.
+      #
+      def _validate_on_demand_resources(h)
+        category = h.dig(h.keys.first, :category)&.to_sym
```

Comment:
this is actually wrong, we need to collect all `:category` from all entries. will fix and update the test.

Reply:
fixed and test updated. pushed.